src/controllers/signInController.js

```diff
@@ -0,0 +1,32 @@
+const { ErrEmailNotFound, ErrInvalidPassword } = require('../services/usersService');
+const BaseController = require('./baseController');
+
+class SignInController extends BaseController {
+  constructor(args) {
+    super();
+    this.usersService = args.usersService;
+  }
+
+  async execute(req, res) {
+    const { email, password } = req.body;
+
+    let user;
+
+    try {
+      user = await this.usersService.signInUser(email, password);
+    } catch (err) {
+      if (err instanceof ErrEmailNotFound || err instanceof ErrInvalidPassword) {
+        return res.send(err.message);
+      } else {
+        console.log(`failed to sign in user: ${JSON.stringify(err)}`);
+        return res.send('Internal server error');
+      }
+    }
+
+    req.session.userId = user.id;
+
+    return res.send('You are signed in.');
+  }
+}
+
+module.exports = SignInController;
```

Comment:
We could make this a little more flexible, but I s'pose it's fine for these controllers to do this. Unless we want uniformity, in which case we'd want:

```js
module.exports = { SignInController };
```

Reply:
ok Mr. M, i'll do it